cmd/scCount/scCount.go

```diff
@@ -0,0 +1,165 @@
+// Command Group: "SAM Tools"
+
+//TODO: Input-normalization.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/common"
+	"github.com/vertgenlab/gonomics/dna"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fileio"
+	"github.com/vertgenlab/gonomics/gtf"
+	"github.com/vertgenlab/gonomics/interval"
+	"github.com/vertgenlab/gonomics/sam"
+	"io"
+	"log"
+	"sort"
+	"strings"
+)
+
+func scCount(s Settings) {
+	var sc sam.SingleCellAlignment
+	var err error
+	readChan, _ := sam.GoReadToChan(s.InFile)
+	genes := gtf.Read(s.GeneFile)
+	var geneIntervals []interval.Interval
+	var geneIdSlice []string = make([]string, 0)
+	var geneIndex = make(map[string]int)
+	var headerString string = "Bx"
+
+	for g := range genes {
+		geneIdSlice = append(geneIdSlice, g)
+	}
+	sort.Strings(geneIdSlice)
+	for c, g := range geneIdSlice { //c for count, g for gene
+		headerString += fmt.Sprintf("\t%s", g)
+		geneIndex[genes[g].GeneID] = c
+		geneIntervals = append(geneIntervals, genes[g])
+		c++
```

Comment:
I have not seen increments mixed with "range."  Is this to do only the odd genes?